### PR TITLE
Replace FileDeleteVisitor with Guava MoreFiles.deleteRecursively()

### PR DESCRIPTION
**Pull request by elharo**
Replace the hand-crafted directory deletion code with Guava 21.0's `MoreFiles.deleteRecursively()`.

@chanseokoh @loosebazooka I think I found the missing piece that caused this to fail on Windows the last time we tried this.

tldr; deleting directories has been a hole in Java's cross-platform story for 20 years and I trust the Guava team to have put more time and effort into thinking out the corner cases and working around them than we have.

### CODE REVIEW

**Comment by patflynn** [comment]
@elharo is there a bug to link this PR to?

**Comment by elharo** [comment]
I don't think so, just a TODO

**Comment by patflynn** [comment]
In that case perhaps just add more info about what this PR fixes to the change description?

**Comment by chanseokoh** [comment]
FTR: https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/2880#issuecomment-370256930

**Comment by briandealwis** [comment]
Updated description

**Comment by elharo** [comment]
tldr; deleting directories has been a hole in Java's cross-platform story for 20 years and I trust the Guava team to have put more time and effort into thinking out the corner cases and working around them than we have.
